pkg/opensearch/response_parser.go

```diff
@@ -285,10 +303,101 @@ func processTraceSpansResponse(res *client.SearchResponse, queryRes backend.Data
 	}
 	frame.Meta.PreferredVisualization = data.VisTypeTrace
 
-	queryRes.Frames = data.Frames{frame}
+	queryRes.Frames = append(queryRes.Frames, frame)
 	return queryRes
 }
 
+func createServiceStatsMap(spanServiceStats []interface{}) map[string]interface{} {
+	serviceMap := make(map[string]interface{})
+	for _, service := range spanServiceStats {
+		serviceName := service.(map[string]interface{})["key"].(string)
+		serviceMap[serviceName] = service
+	}
+	return serviceMap
+}
+
+// processServiceMapResponse combines information from the ServiceMap and Stats requests to build
+// dataframes the NodeGraph panel will use to display the service map. We send one frame for edge
+// information with source, target and a name (in this case just "<source>_<target>", and one
+// for node information with the service name, latency, throughput, error info, etc.
+func processServiceMapResponse(serviceMap []interface{}, spanServiceStats []interface{}, duration time.Duration, traceId string) data.Frames {
+	edgeFields := Fields{}
+	edgeIds := edgeFields.Add("id", nil, []string{})
+	edgeSources := edgeFields.Add("source", nil, []string{})
+	edgeDestinations := edgeFields.Add("target", nil, []string{})
+	edgeDetails := edgeFields.Add("detail__operation", nil, []string{}, &data.FieldConfig{DisplayName: "Operation(s)"})
+
+	nodeFields := Fields{}
+	nodeIds := nodeFields.Add("id", nil, []string{})
+	nodeTitles := nodeFields.Add("title", nil, []string{}, &data.FieldConfig{DisplayName: "Service name"})
+	nodeErrorRates := nodeFields.Add("arc__errors", nil, []float64{}, &data.FieldConfig{Color: map[string]interface{}{"mode": "fixed", "fixedColor": "red"}})
+	nodeErrorRatesDetails := nodeFields.Add("detail__errors", nil, []float64{}, &data.FieldConfig{DisplayName: "Error rate", Unit: "%"})
+	nodeSuccessRates := nodeFields.Add("arc__success", nil, []float64{}, &data.FieldConfig{Color: map[string]interface{}{"mode": "fixed", "fixedColor": "green"}})
```

Comment:
Should `arc__errors` and `arc__success` have display names as well? I see them in the legend in the bottom left hand corner of the panel.